p/db/index.js

```diff
@@ -186,26 +184,97 @@ Pass in just opt.age, and the age group will calculated automatically.`)
 
 	const tvlrAgeGroup = ('age' in opt) ? ageGroupFromAge(opt.age) : opt.ageGroup
 
-	query.trfReq = {
-		// todo: what are these?
-		// "directESuiteCall": true,
-		// "rType": "DB-PE",
-
+	const basicCtrfReq = {
 		jnyCl: opt.firstClass === true ? 1 : 2,
 		// todo [breaking]: support multiple travelers
 		tvlrProf: [{
 			type: tvlrAgeGroup || ageGroup.ADULT,
 			...(('age' in opt) ? {age: opt.age} : {}),
-			redtnCard: opt.loyaltyCard
-				? formatLoyaltyCard(opt.loyaltyCard)
-				: null
+			redtnCard: opt.loyaltyCard ? formatLoyaltyCard(opt.loyaltyCard) : null,
 		}],
 		cType: 'PK'
 	}
+	if ((refreshJourney) && (opt.tickets)) {
+		// todo: what are these?
+		// basicCtrfReq.directESuiteCall = true
+		// If called with "Reconstruction"
+		// 'DB-PE' causes the response to contain the tariff information.
+		basicCtrfReq.rType = 'DB-PE'
+	}
+	return basicCtrfReq
+}
+
+const transformJourneysQuery = ({opt}, query) => {
+	const filters = query.jnyFltrL
+	if (opt.bike) filters.push(bike)
+	query.trfReq = trfReq(opt, false)
 
 	return query
 }
 
+const formatRefreshJourneyReq = (ctx, refreshToken) => {
+	const {profile, opt} = ctx
+	const req = {
+		getIST: true,
+		getPasslist: !!opt.stopovers,
+		getPolyline: !!opt.polylines,
+		getTariff: !!opt.tickets
+	}
+	if (profile.refreshJourneyUseOutReconL) {
+		req.outReconL = [{ctx: refreshToken}]
+	} else {
+		req.ctxRecon = refreshToken
+	}
+	req.trfReq = trfReq(opt, true)
+
+	return {
+		meth: 'Reconstruction',
+		req,
+	}
+}
+
+
+const addDbOfferSelectionUrl = (journey, opt) => {
+
+	// if no ticket contains addData, we can't get the offer selection URL
+	if (journey.tickets.some((t) => t.addDataTicketInfo)) {
+
+		const queryParams = new URLSearchParams();
+
+		// Add individual parameters
```

Comment:
Just for future investigations: This is closely related to, but not the same as, https://github.com/derhuerst/generate-db-shop-urls/blob/e9f5d4385eb5985d52f244921bb7562efc3d7d64/index.js#L99-L136.